TEST_APPS/device/nfcapp/nfccommands.cpp

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <string>
+#include <vector>
+#include <stdlib.h>
+#include "mbed_events.h"
+#include "mbed-client-cli/ns_cmdline.h"
+#include "rtos\Thread.h"
+#include "nfcTestShim.h"
+#include "nfcCommands.h"
+
+#if MBED_CONF_NFCEEPROM
+#include "NFCEEPROMDriver.h"
+#include "nfcProcessEeprom.h"
+#endif
+
+using mbed::nfc::nfc_rf_protocols_bitmask_t;
+
+events::EventQueue nfcQueue;
+rtos::Thread nfcThread;
+NFCTestShim *pNFC_Test_Shim = NULL;
+
+NFCTestShim *new_testshim()
+{
+#if MBED_CONF_NFCEEPROM
+    mbed::nfc::NFCEEPROMDriver &eeprom_driver = get_eeprom_driver(nfcQueue);
+
+    return ((NFCTestShim *)(new NFCProcessEEPROM(nfcQueue, eeprom_driver)));
+#else
+    return ((NFCTestShim *)(new NFCProcessController(nfcQueue)));
+#endif // EEPROM
+
+}
+
+void nfcRoutine()
+{
+    nfcQueue.dispatch_forever();
+}
+
+HandleTestCommand::HandleTestCommand()
+{
+    osStatus status = nfcThread.start(mbed::callback(&nfcRoutine));
```

Comment:
I'm not a big fan of using globals within classes unless there's a strong reason for doing it - could `nfcRoutine()` be a member of `HandleTestCommand`?